packages/provider/src/api/verify.ts

```diff
@@ -0,0 +1,204 @@
+// Copyright 2021-2024 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {
+    ApiParams,
+    ApiPaths,
+    ImageVerificationResponse,
+    ServerPowCaptchaVerifyRequestBody,
+    VerificationResponse,
+    VerifySolutionBody,
+    VerifySolutionBodyType,
+} from '@prosopo/types'
+import { CaptchaStatus } from '@prosopo/captcha-contract/types-returns'
+import { ProsopoApiError } from '@prosopo/common'
+import { ProviderEnvironment } from '@prosopo/types-env'
+import { Tasks } from '../tasks/tasks.js'
+import { getBlockTimeMs, getCurrentBlockNumber } from '@prosopo/contract'
+import { verifySignature } from './authMiddleware.js'
+import express, { Router } from 'express'
+
+/**
+ * Returns a router connected to the database which can interact with the Proposo protocol
+ *
+ * @return {Router} - A middleware router that can interact with the Prosopo protocol
+ * @param {Environment} env - The Prosopo environment
+ */
+export function prosopoVerifyRouter(env: ProviderEnvironment): Router {
+    const router = express.Router()
+    const tasks = new Tasks(env)
+
+    /**
+     * Verifies a user's solution as being approved or not
```

Comment:
Use docstrings to explain what endpoints are doing. 

This docstring is wrong - dapp not user.

Reply:
I have done